### PR TITLE
Refactor tests

### PR DESCRIPTION
**Pull request by dajenet**
Make tests transactional.

#32

### CODE REVIEW

**Comment by michael-kamel** [comment]
maybe we can use [java-faker](https://www.baeldung.com/java-faker)

File: src/test/java/science/icebreaker/util/TestHelper.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        this.deviceAvailabilityRepository = deviceAvailabilityRepository;
+    }
+
+    private Account createAccountInternal(RegistrationRequest request) {
+        doReturn(null).when(mailService).sendMail(anyString(), anyString(), anyString());
+
+        int accountId = accountService.createAccount(request);
+        Account account = request.getAccount();
+        account.setId(accountId);
+        accountService.enableAccount(account.getEmail());
```
Comment (michael-kamel) [inline src/test/java/science/icebreaker/util/TestHelper.java]: `account.setEnabled(true)` should do the trick

**Review by michael-kamel — COMMENTED** [review]
I could suggest using a library to generate data instead of having multiple functions for the data change but then this could make it a little bit harder to debug errors in tests.

Maybe we can also have the tests as per method instead and drop the `@Transactional`, this could be more useful for integration tests instead.

**Comment by dajenet** [comment]
I would like to keep the tests deterministic. I think the number of different objects is low enough to create them manually.

**Review by michael-kamel — APPROVED** [review]

**Review by michael-kamel — APPROVED** [review]

**Pull request merged**